### PR TITLE
MULE-10277: Implement error mappings for message processors

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

**Comment by afelisatti** [comment]
REVIEW ONLY.

**Comment by afelisatti** [comment]
All tests have passed on CI.

File: modules/dsl-api/src/main/java/org/mule/runtime/dsl/api/component/AbstractAnnotatedObjectFactory.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.mule.runtime.api.meta.AnnotatedObject;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.xml.namespace.QName;
+
+public abstract class AbstractAnnotatedObjectFactory<T> implements AnnotatedObjectFactory<T> {
```
Comment (pablolagreca) [inline modules/dsl-api/src/main/java/org/mule/runtime/dsl/api/component/AbstractAnnotatedObjectFactory.java]: javadoc

File: modules/dsl-api/src/main/java/org/mule/runtime/dsl/api/component/AnnotatedObjectFactory.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.mule.runtime.api.meta.AnnotatedObject;
+
+/**
+ * Interface that must be implemented by those classes that are meant to be used as a factory to create complex domain objects
+ * which in turn are {@link AnnotatedObject}s.
+ *
+ * @param <T> the type of the object to be created, which should be an {@link AnnotatedObject}.
+ */
+public interface AnnotatedObjectFactory<T> extends ObjectFactory<T>, AnnotatedObject {
```
Comment (pablolagreca) [inline modules/dsl-api/src/main/java/org/mule/runtime/dsl/api/component/AnnotatedObjectFactory.java]: add in the javadoc a reference to the AbstractAnnotatedObjectFactory as helper class

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java [hunk 3]
```diff
@@ -503,6 +506,25 @@ private boolean isMuleConfigurationFile() {
     return muleComponentModels.get(0).getIdentifier().equals(MULE_IDENTIFIER);
   }
 
+  private void validateErrorMappings() {
+    executeOnEveryComponentTree(componentModel -> {
+      List<ComponentModel> errorMappings = componentModel.getInnerComponents().stream()
+          .filter(c -> c.getIdentifier().equals(ERROR_MAPPING_IDENTIFIER)).collect(toList());
+      if (!errorMappings.isEmpty()) {
```
Comment (pablolagreca) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java]: What about repeated mappings for the same source error type?

Comment (afelisatti) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java]: I've added a validation and test for that in the latest commit

Comment (afelisatti) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java]: Also, I realised I could have a general test using the PetStoreConnector so I've included that as well

**Pull request merged**